app/services/customers/update_service.rb

```diff
@@ -190,6 +194,10 @@ def call
     attr_reader :customer, :args
     def_delegators :customer, :organization
 
+    def billing_entity
+      @billing_entity ||= organization.billing_entities.active.find_by!(code: args[:billing_entity_code])
```

Comment:
I'm thinking: should we update billing_entity relation on organization, to not have to specify `active` each time... because usually we DO have only active...
```
  has_many :billing_entities, -> { where(archived_at: nil) }
  has_many :all_billing_entities, class_name: "BillingEntity"
```